### PR TITLE
Automatic class installation

### PR DESCRIPTION
**Pull request by damonge**
first version, working in C, not in python yet

### CODE REVIEW

**Comment by damonge** [comment]
Alright, this should be working now. There's a number of things to check:
- [ ] That travis passes
- [ ] That you can follow the new instructions in README.md regarding the CLASS installation, and that those instructions are sufficiently clear.
- [ ] That the class_install.py script runs without any problems on both linux and mac
- [ ] That the C library can be installed in both linux and mac
- [ ] That the python module can be installed in both linux and mac
- [ ] That the C and python unit tests pass.

Obviously I've checked all of these (at least in linux). But it'd be good if the reviewer could verify them.

**Comment by rainwoodman** [comment]
I had a glance at `.travis.yaml` What about setting it up to use miniconda which comes with gsl and is expected to work on osx? - I suspect it's going to be the way most people get Python on osx anyways.

**Comment by fjaviersanchez** [comment]
@rainwoodman Does miniconda include GSL2x or just GSL1x?

**Comment by fjaviersanchez** [comment]
@damonge the tests keep failing because it is not finding the CCL library nor the CLASS library. I'll take a look at the install script to see what's going on.

**Comment by rainwoodman** [comment]
https://anaconda.org/anaconda/gsl

It looks like 2.0.

**Comment by fjaviersanchez** [comment]
@damonge `wget` is not default in MAC so maybe we should switch to `curl` or try `git clone` from the class repository?

**Comment by fjaviersanchez** [comment]
@damonge I solved the `wget` part but also MacOS X is not compatible with openmp by default. I am trying to change that as well

**Comment by damonge** [comment]
@fjaviersanchez @elisachisari @philbull : OK, travis checks are passing now. Can you guys check that you can install the library following the new procedure described in the README? It'd be good to check that things work at least in linux and mac

**Comment by fjaviersanchez** [comment]
@damonge It installs beautifully in my laptop!

**Comment by elisachisari** [comment]
@damonge. I tried installing in the usual way in my Mac, and got the following problem:
screen shot 2017-12-20 at 10 09 11 am

File: README.md [hunk 1]
```diff
@@ -23,9 +23,15 @@ In order to compile `CCL` you need a few libraries:
...
 * [FFTW3](http://www.fftw.org/) is required for computation of correlation functions.
 * FFTlog([here](http://casa.colorado.edu/~ajsh/FFTLog/) and [here](https://github.com/slosar/FFTLog))is provided within `CCL`, with minor modifications.
+* The C library associated to the CLASS code. The installation of this library is described below.
+
+## Installing CLASS
+CCL uses CLASS as one of the possible ways of computing the matter power spectrum. In order to communicate with CLASS, CCL must be linked to its library. Before installing CCL proper you must therefore install this library first. Since this process is not necessarily straightforward, we provide a python script `class_install.py` that automatically downloads and install the latest tagged stable version of CLASS. You should run this script (`python class_install.py`) before carrying out the next steps. By default, the script assumes that your main C compiler is `gcc`. If that's not the case, pass the name of your C compiler to the script via the command-line argument `--c_comp` (i.e. `python class_install.py --c_comp=[name of compiler]`). Type `python class_install.py -h` for further details.
```
Comment (elisachisari) [inline README.md]: I think this script downloads a specific version of CLASS - v2.6.3. Do we want to give instructions to the user to potentially replace that with a newer version when available?

Comment (damonge) [reply README.md]: I'd say we should be in charge of keeping that up to date. If they want their own version of CLASS they should follow the steps for the custom CLASS install.

Comment (elisachisari) [reply README.md]: Ok

**Comment by elisachisari** [comment]
If I start from scratch in a new CCL folder which does not have a previous installation, everything works fine. For any new user downloading CCL after this merge, there won't be problems. I suggest adding a comment in the README that if you have a previous installation of CCL which didn't have the CLASS patch, you might want to start from scratch by cloning a new repo. Or you could run the CLASS installation twice as I did.

**Comment by damonge** [comment]
Sounds good, I'll do that

**Comment by elisachisari** [comment]
@damonge - could you also edit the CCL note to match these changes before you merge?

**Comment by damonge** [comment]
Note modified to describe the new CLASS installation

File: README.md (hunk 1 above)
Comment (jchiang87) [reply README.md]: Should the section "Compiling against an external version of CLASS" be omitted or revised?

Comment (damonge) [reply README.md]: Not really. These modifications automatically download and install a tagged version of CLASS that we have chosen. If users want to use any other CLASS (e.g. hi-CLASS), they have to follow the instructions in that section

Comment (jchiang87) [reply README.md]: It still says "CCL has a built-in version of CLASS" (the first sentence of that section).  I guess I'm missing something, but I thought the built-in version has been removed.

Comment (damonge) [reply README.md]: Oh, I see. I'll fix that quickly. Can you review a quick PR?

**Pull request merged**